### DISCUSSION TITLE
Problems with Related categories list

### DISCUSSION THREAD

**Discussion by liquid-a** [codefog/contao-news_categories#219] [Q&A] [answered]
The documentation states that the following is possible: 

Related categories list

The news list module has also a feature to display the related news items based on the categories of the current item. This is typically needed when you have a news reader module on the page and below you would like to display similar news. To achieve this, the module structure on the page has to be the following:

news reader module
news list module (in related categories mode)

But when I select this option, I only get the output "There are currently no messages available" below the news reader module.

Am I missing a certain setting, or do I have to select something special? If I don't select the Related categories list, I get news, but unfortunately all mixed and none matching the category. I use Contao 4.13

**Comment by qzminski** [comment]
Can you post a screenshot of the backend settings of both mentioned frontend modules?

**Comment by liquid-a** [comment]
Here I send the two screenshots. While I was making these, I changed something again and suddenly the two news items were below the news reader. I had selected a customised template in the module template "mod_newsreader". This does not seem to work. Now there were 2 news under the news reader. 
But I thought, as stated in the instructions, you need both a newsreader module and a news list module below it, both on one page. Now it seems that this is only integrated in the newsreader module. 

I would prefer to be able to place a own news list under the newsreader, which then generates an output as described in the instructions. Currently, I can only display a similar output from the archive, but I would like to create an output sorted by category (contao-news_categories).

Below are the screenshots before the last adjustment:

![screenshot_reader](https://user-images.githubusercontent.com/7712553/188418711-db54efa2-b446-4170-a74b-b82f1a4ee54c.jpg)

![screenshot_newslist_below_reader](https://user-images.githubusercontent.com/7712553/188418729-76e7973b-e740-4129-8cff-8eeedfb8f699.jpg)

**Comment by qzminski** [comment]
The news reader module has no effect on the related news list module. I would rather inspect your custom templates if they contain some stuff respective for this problem.

**Comment by liquid-a** [comment]
If I select the default template for both in the news reader and in the news list below the news reader (everywhere), I also get the output: "There are currently no messages available".
Unfortunately, this is not the solution either.

**Comment by qzminski** [comment]
Can you share your custom templates, please?

**Comment by liquid-a** [comment]
Here the Template news_full.html5:
![news_full](https://user-images.githubusercontent.com/7712553/188592269-ff5929e2-9459-4207-9b03-2a0fb12e5264.jpg)

**Comment by liquid-a** [comment]
And the news_latest from the newslist directly under the news reader:
![news_latest](https://user-images.githubusercontent.com/7712553/188592662-d5ea421c-19a8-4494-804f-7a00311332e1.jpg)

**Comment by qzminski** [comment]
Have you got any custom `mod_newsreader` or `mod_newslist` templates?

**Reply by liquid-a** [reply]
No, here I use the standard Contao templates, without any customisation.
I can also send you access data to the Contao website, then this might be easier to keep track of. Can I send this to kamil.kuzminski@codefog.pl ?

**Comment by qzminski** [comment, marked as answer]
@liquid-a I have checked you environment and your configuration is incorrect:

1. You use the `news_related` module is NOT a part of News Categories extension. You should use the news list with the `Related categories mode` checkbox selected.
2. You also have some strange modules like `Newslist Infinite Scroll` which may have an impact on this functionality as well.

I have created the `! Newslist related` frontend module which is the news list with the `Related categories mode` checkbox selected and it works now.

**Reply by qzminski** [reply]
Also, to clarify: the option in the news reader to display the related articles is NOT a part of News Categories extension either. It must come from a different extension, probably the one that also adds `news_related` frontend module.

**Comment by liquid-a** [comment]
Many thanks for the help and the solution to the problem! I must have made a mistake and had too many modules in use for the news.
